src/main/java/com/velox/sloan/cmo/workflows/planning/CustomFieldsRetriever.java

```diff
@@ -0,0 +1,464 @@
+     */
+    private Integer getCoverageTargetForsample(DataRecord sample) throws IoError, RemoteException, NotFound, ServerException, InvalidValue {
+        String sequencingRequirementDatatype = getSequencingRequirementDataType(sample);
+        logInfo("SeqRequirement DataType: " + sequencingRequirementDatatype);
+        Integer coverageTarget = null;
+        DataRecord[] sequencingRequirementRecords = sample.getChildrenOfType(sequencingRequirementDatatype, user);
+        if (sequencingRequirementRecords.length > 0 && sequencingRequirementRecords[0].getValue("CoverageTarget", user) != null) {
+            coverageTarget = sequencingRequirementRecords[0].getIntegerVal("CoverageTarget", user);
+        } else {
+            boolean found = false;
+            DataRecord startingSample = sample;
+            while (!found && startingSample != null) {
+                List<DataRecord> parentSamples = startingSample.getParentsOfType("Sample", user);
+                DataRecord parentSample = null;
+                if (!parentSamples.isEmpty()) {
+                    parentSample = parentSamples.get(0);
+                }
+                if (parentSample != null && parentSample.getChildrenOfType(sequencingRequirementDatatype, user).length > 0
+                        && parentSample.getChildrenOfType(sequencingRequirementDatatype, user)[0].getValue("CoverageTarget", user) != null) {
+                    coverageTarget = parentSample.getChildrenOfType(sequencingRequirementDatatype, user)[0].getIntegerVal("CoverageTarget", user);
+                    found = true;
+                } else {
+                    startingSample = parentSample;
+                }
+            }
+        }
+        if (coverageTarget != null && coverageTarget > 0) {
+            return coverageTarget;
+        } else {
+            clientCallback.displayWarning(String.format("Invalid Target Coverage '%d' for sample '%s'.\nPlease double check.", coverageTarget, sample.getStringVal("SampleId", user)));
+            return 0;
+        }
+    }
+
+    /**
+     * Because the sequencing requirements for a Library Sample and Pooled Library Sample are stored under different DataRecords,
+     * this method returns the DataType name for DataRecord containing Sequencing Requirement values for a sample.
+     * @param sample
+     * @return String value DataType name for DataRecord containing Sequencing Requirement values for a sample.
+     * @throws NotFound
+     * @throws RemoteException
+     * @throws ServerException
+     * @throws InvalidValue
+     */
+    private String getSequencingRequirementDataType(DataRecord sample) throws NotFound, RemoteException, ServerException, InvalidValue {
+        String sampleType = sample.getStringVal("ExemplarSampleType", user);
+        logInfo("SampleType: " + sampleType);
+        if (sampleType.toLowerCase().equals(librarySampleTypes.get(0)) || sampleType.toLowerCase().equals(librarySampleTypes.get(1))) {
```

Comment:
how about using equalsignorecase

Reply:
Changed all such statement to as suggested.